src/fs/at.rs

```diff
@@ -82,7 +84,8 @@ fn _readlinkat(dirfd: BorrowedFd<'_>, path: &ZStr, mut buffer: Vec<u8>) -> io::R
             buffer.resize(nread, 0_u8);
             return Ok(ZString::new(buffer).unwrap());
         }
-        buffer.resize(buffer.len() * 2, 0_u8);
+        buffer.reserve(buffer.capacity()); // double capacity
```

Comment:
I agree, would also work.
Here, I just wanted to make sure that the growth is always exponential (independent from the underlying `reserve` implementation).
However, since the problem `Vec` is trying to solve (reducing the number of `realloc` syscall invocations) is very similar to limiting the number of `readlinkat` syscalls, the Vec resizing behavior will probably always be optimal.
I'll update the code and add a comment with some explanation.